corgi/core/models.py

```
@@ -488,6 +522,23 @@ def save_product_taxonomy(self):
         # All ProductModels have a set of descendant channels
         self.channels.set(channels)
 
+    def get_related_names_of_type(self, mapping_model: type) -> list[str]:
+        """For a given ProductModel instance find all directly related nodes with the given model type
+        and return their names"""
+        mapping_key = mapping_model.__name__
+        # No .distinct() since __name on all ProductModel subclasses + Channel is always unique
+        attribute_name = mapping_key.lower()
+        pnode = self.pnodes.exclude(type=ProductNode.ProductNodeType.INFERRED).first()
```

In the old code, I used .get() to enforce that models had exactly one node. There are probably other places with this assumption that will need to be fixed.

For this code here, I get that we sometimes expect a Variant to have multiple nodes now. But can we raise an error if no node is found at all?

It's also helpful if you can edit the existing code instead of deleting the old code and adding the new code, so that the diff is easier to parse. You can do the editing in one commit, then move the method in another commit, so commit-by-commit reviewing still works without having to jump around to compare stuff.

---

We don't want to return an error, we just don't want to return anything.

---

Why not though? Don't we expect a Variant (whether direct or inferred) to always have at least one node?